includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -531,6 +531,16 @@ public function should_show_payment_request_button() {
 			return false;
 		}
 
+		// Cart total is 0 or is on prouct page and product price is 0.
+		if (
+			( 0.0 === (float) WC()->cart->get_total( 'edit' ) && ! $this->is_product() ) ||
+			( $this->is_product() && 0.0 === (float) $this->get_product()->get_price() )
+
+		) {
+			Logger::log( 'Order price is 0 ( Payment Request button disabled )' );
```

Comment:
I wonder if we should include more identifiable information here, eg. product id/name.

Reply:
I wouldn't rely on products, as this could be on checkout page, and we could have more than 1 product with price 0.

Reply:
Maybe some other identifier? I'm trying to visualize how it will look like in the logs while investigating a potential issue.

Reply:
changed in [054f511](https://github.com/Automattic/woocommerce-payments/pull/6334/commits/054f5117b778927591491227c68d1f99aa6a6961) . thx

Reply:
@dpaun1985 Thanks for the change. On another thought, we might not need it based on our discussion in our team meeting. Sorry about the confusion, but can we undo this change?

Reply:
done.